File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/Tv.java
/*
 * Copyright (c) 2018 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models;

import android.support.annotation.NonNull;

import java.io.Serializable;
import java.util.Arrays;

/** テレビリモコンの情報 */
public class Tv implements Serializable {
    /** ボタンリスト */
    public Button[] buttons;

    public State state;

    public static class Button implements Serializable {
        /** ボタン名 */
        public String name;
        /** ボタンアイコン */
        public String image;
        /** ラベル */
        public String label;

        @NonNull
        @Override
        public String toString() {
            return "Button{" + "name='" + name + '\'' + ", image='" + image + '\'' + ", label='" + label + '\'' + '}';
        }
    }

    public static class State implements Serializable {
        /**
         * TODO APIリファレンスに無いので分からない
         * TOSHIBAのリモコンで「t」が取得できた
         * ボタン表示のレイアウトの識別記号？
         */
        public String input;

        @NonNull
        @Override
        public String toString() {
            return "State{" + "input='" + input + '\'' + '}';
        }
    }

    @NonNull
    @Override
    public String toString() {
        return "TV{" + "buttons=" + Arrays.toString(buttons) + ", state=" + state + '}';
    }
}

File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/Tv.kt
/*
 * Copyright (c) 2019 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fishwaffle.natureremo.controller.NatureRemo.isIgnoreUnknown
import java.io.Serializable

/** テレビリモコンの情報
 * @param buttons ボタンリスト
 * @param state 状態
 * */
@JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
data class Tv(var buttons: List<Button>? = null, var state: State? = null) : Serializable {

    /**
     * TVボタン情報
     * @param name ボタン名
     * @param image ボタンアイコン
     * @param label ラベル
     */
    @JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
    data class Button(var name: String?, var image: String?, var label: String?) : Serializable

    /**
     * TVStatus
     * @param input [t, bs, cs]
     */
    @JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
    data class State(var input: String? = null) : Serializable

}
